src/App.tsx

```diff
@@ -1,24 +1,71 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React,
+{
+  useCallback, useContext, useEffect, useMemo,
+} from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import * as todoService from './api/todos';
+import { Todo } from './types/Todo';
+import { FilterBy } from './types/FilterBy';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { ErrorType } from './types/ErrorType';
+import { ErrorInfo } from './components/ErrorInfo/ErrorInfo';
+import { filterTodos } from './helpers/filterTodos';
+import { USER_ID } from './utils/userId';
+import { AppContext } from './contexts/appContext';
 
 export const App: React.FC = () => {
+  const {
+    todosFromServer,
+    filterBy,
+    setTodosFromServer,
+    setErrorMessage,
+  } = useContext(AppContext);
+
+  useEffect(() => {
+    todoService.getTodos(USER_ID)
+      .then(setTodosFromServer)
+      .catch(() => setErrorMessage(ErrorType.UnableToLoad));
+  }, [setTodosFromServer, setErrorMessage]);
+
+  const memoizedFilterTodos = useCallback(
+    (
+      todos: Todo[],
+      filterByOption: FilterBy,
+    ) => filterTodos(todos, filterByOption),
+    [],
+  );
+
+  const todosToView = useMemo(
+    () => memoizedFilterTodos(todosFromServer, filterBy),
```

Comment:
```suggestion
  const todosToView = useMemo(
    () => filterTodos(todosFromServer, filterBy),
```
`memoizedFilterTodos` is redundant as it just wrap `filterTodos`